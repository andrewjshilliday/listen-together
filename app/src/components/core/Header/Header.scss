.header {
  display: flex;
  background-color: var(--background-darker);
  color: var(--on-background);
  padding: 0 25px;
  border-bottom: 1px solid black;
  height: 75px;

  h1 {
    margin: 0;
    line-height: 75px;
  }
}

.logo {
  margin: auto 0;
  font-size: 2em;
  font-weight: 500;
}

.auth-button {
  margin: auto 0;
}

.search-field {
  margin: auto 0 auto auto;
  padding-right: 25px;
  position: relative;
  top: -8px;
}
