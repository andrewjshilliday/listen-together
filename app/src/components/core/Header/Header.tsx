import React, { useState } from 'react';
import { useHistory, Link } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { Button, IconButton, TextField, InputAdornment } from '@material-ui/core';
import SearchIcon from '@material-ui/icons/Search';
import { ApplicationState } from '../../../store';
import { signIn, signOut } from '../../../store/authentication';
import styled from 'styled-components';
import musicSvg from '../../../assets/images/listen-together.svg';

const Header: React.FC = (props: any) => {
  const dispatch = useDispatch();
  const authenticated = useSelector((state: ApplicationState) => state.authentication.authenticated);
  const roomId = useSelector((state: ApplicationState) => state.room.roomId);
  const history = useHistory();
  const [searchQuery, setSearchQuery] = useState('');

  const search = () => {
    if (searchQuery) {
      history.push(`/search/${searchQuery}`);
    }
  }
  
  let authButton, searchField;
  if (authenticated) {
    authButton = <Button variant="contained" color="primary" onClick={() => dispatch(signOut())}>Sign Out</Button>;
    searchField = 
      <TextField
        label="Search"
        onChange={(e) => setSearchQuery(e.target.value)}
        onKeyPress={(e) => {
          if (e.key === 'Enter') {
            e.preventDefault();
            search();
          }
        }}
        InputProps={{
          endAdornment: (
            <InputAdornment position="end">
              <IconButton onClick={() => search()}>
                <SearchIcon />
              </IconButton>
            </InputAdornment>
          )
        }}
      />;
  } else {
    authButton = <Button variant="contained" color="primary" onClick={() => dispatch(signIn())}>Sign In</Button>;
  }
  
  return (
    <HeaderContainer>
      <Logo className="text-truncate" to={roomId ? `/room/${roomId}` : `/`}>
        <img src={musicSvg} alt="ListenTogether" /> ListenTogether
      </Logo>
      {roomId && <SearchField>{searchField}</SearchField>}
      <AuthButton>{authButton}</AuthButton>
    </HeaderContainer>
  );
}

export default Header;


const HeaderContainer = styled.div`
  display: flex;
  align-items: center;
  background-color: var(--background-darker);
  color: var(--on-background);
  padding: 0 25px;
  border-bottom: 1px solid var(--on-background);
  height: 75px;

  h1 {
    margin: 0;
    line-height: 75px;
  }
`;
const Logo = styled(Link)`
  margin-right: auto;
  font-size: 2em;
  font-weight: 500;

  img {
    width: 65px;
    height: 43px;
  }
`;
const SearchField = styled.div`
  padding-right: 25px;
  position: relative;
  top: -8px;
`;
const AuthButton = styled.div``;
