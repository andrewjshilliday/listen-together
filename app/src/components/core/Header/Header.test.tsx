import React from 'react';
import { StaticRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from './Header';

jest.mock('../../providers/AuthorizationProvider', () => require('../../../__mocks__/providers/AuthorizationProvider.mock').default);
jest.mock('../../providers/RoomProvider', () => require('../../../__mocks__/providers/RoomProvider.mock').default);

it('renders correctly', () => {
  const tree = renderer.create(
    <StaticRouter location=".">
      <Header />
    </StaticRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});
