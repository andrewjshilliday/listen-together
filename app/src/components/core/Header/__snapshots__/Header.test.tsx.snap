// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxjAm jbAssD"
>
  <a
    className="sc-AxirZ iIFQSL text-truncate"
    href="/room/TEST1"
    onClick={[Function]}
  >
    <img
      alt="ListenTogether"
      src="listen-together.svg"
    />
     ListenTogether
  </a>
  <div
    className="sc-AxiKw bRYEjz"
  >
    <div
      className="MuiFormControl-root MuiTextField-root"
      onKeyPress={[Function]}
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
      >
        Search
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
        <div
          className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-AxhCb ccMvAG"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Sign Out
      </span>
    </button>
  </div>
</div>
`;
